### DISCUSSION TITLE
http device: overlap with sandstorm bridge?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#11222] [General]
who just suggested looking into this in FriAm? Was it @kpreid ?

**Comment by kpreid** [comment]
Yep. I was thinking that both projects could benefit from not separately designing and refining "turn external HTTP into structured capability-flavored messages and back".

**Comment by erights** [comment]
Hi @kpreid there are enough semantic differences that I don't think we can reconcile with cap'n proto directly. But I would like to see a bridge/adapter built. This is probably most relevant to the issues @cwebber is paying attention to, of trying to unify the various captp family protocols, or at least interoperate between them.

Assigning it to myself just to track and remind for now.

**Comment by dckc** [comment]
The http device is part of the ag-solo architecture, which is not used in production.

Let's migrate this to a discussion.
